lib/_http_parser.js

```diff
@@ -0,0 +1,890 @@
+/*
```

Comment:
> this is deprecating process.binding('http_parser') which means we're gonna break some people, maybe too many to be able to remove this right away.

`process.binding('http_parser')`, like other bindings, is and always has been internal.  If people are using it directly and removal breaks their code, the onus is on them.

Besides, they can probably substitute it with https://github.com/creationix/http-parser-js.

> if we don't expose this parser for use externally then I think it should be it's own library and vendored in like readable-stream is so that people can require it on npm. if we do that then I think it's reasonable to make it internal.

I'm not sure I follow this logic.  If someone feels strongly enough about it, they can always break out the code into a standalone module.